### ISSUE TITLE
tab autocomplete on user id crashes app

### ISSUE THREAD

**Issue by lancew** [stripedpajamas/scat#32]
When typing in a user id and using tab complete application crashes.
i.e 
```> @squ``` then hitting tab key

This causes a following output (I have removed the actual chat messages):

```
$ npm start -- --debug

> ssb-chat@1.7.1 start /home/lancew/dev/scat
> node index.js "--debug"

Connecting to scuttlebot...
:: PUBLIC MODE ::                                                                                                                                                                                                                              
Sep 14 08:15 LanceW TZAG all, on a train... Scuttlebutt working fine with the intermittent internet! 😃 
...
```

at this point all is working well and I was able to post a message ok. I typed:
`@squ` then hit the TAB key

Which resulted in a crash and the following:

```
> @squ|TypeError: Cannot read property 'startsWith' of null
    at actions.authors.get.filter.name (/home/lancew/dev/scat/node_modules/ssb-chat-core/state/actions.js:77:24)
    at /home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:3084:23
    at /home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1377:16
    at ValueNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1744:12)
    at BitmapIndexedNode.iterate.HashArrayMapNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1737:24)
    at HashArrayMapNode.BitmapIndexedNode.iterate.HashArrayMapNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1737:24)
    at Map.__iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1375:32)
    at KeyedIterable.filterSequence.__iterateUncached (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:3083:16)
    at seqIterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:604:16)
    at KeyedIterable.Seq.__iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:274:14)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ssb-chat@1.7.1 start: `node index.js "--debug"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ssb-chat@1.7.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lancew/.npm/_logs/2018-09-16T09_11_19_288Z-debug.log
```

**Comment by lancew** [comment]
My git log shows me at commit 0d0f8a2f75ed22de09590d9022d538908ee8f45a

debug log below:

`
$ cat /home/lancew/.npm/_logs/2018-09-16T09_11_19_288Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/home/lancew/.nvm/versions/node/v10.0.0/bin/node',
1 verbose cli   '/home/lancew/.nvm/versions/node/v10.0.0/bin/npm',
1 verbose cli   'start',
1 verbose cli   '--',
1 verbose cli   '--debug' ]
2 info using npm@6.4.1
3 info using node@v10.0.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle ssb-chat@1.7.1~prestart: ssb-chat@1.7.1
6 info lifecycle ssb-chat@1.7.1~start: ssb-chat@1.7.1
7 verbose lifecycle ssb-chat@1.7.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle ssb-chat@1.7.1~start: PATH: /home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/lancew/dev/scat/node_modules/.bin:/home/lancew/perl5/bin:/home/lancew/.phpenv/bin:/home/lancew/.nvm/versions/node/v10.0.0/bin:/home/lancew/perl5/perlbrew/bin:/home/lancew/perl5/perlbrew/perls/perl-5.28.0/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/android-sdk/platform-tools:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/opt/flutter/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/lancew/.gem/ruby/2.4.0/bin:/home/lancew/.perl6/bin
9 verbose lifecycle ssb-chat@1.7.1~start: CWD: /home/lancew/dev/scat
10 silly lifecycle ssb-chat@1.7.1~start: Args: [ '-c', 'node index.js "--debug"' ]
11 silly lifecycle ssb-chat@1.7.1~start: Returned: code: 1  signal: null
12 info lifecycle ssb-chat@1.7.1~start: Failed to exec start script
13 verbose stack Error: ssb-chat@1.7.1 start: `node index.js "--debug"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (/home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:947:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:236:5)
14 verbose pkgid ssb-chat@1.7.1
15 verbose cwd /home/lancew/dev/scat
16 verbose Linux 4.18.7-arch1-1-ARCH
17 verbose argv "/home/lancew/.nvm/versions/node/v10.0.0/bin/node" "/home/lancew/.nvm/versions/node/v10.0.0/bin/npm" "start" "--" "--debug"
18 verbose node v10.0.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error ssb-chat@1.7.1 start: `node index.js "--debug"`
22 error Exit status 1
23 error Failed at the ssb-chat@1.7.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
`

**Comment by stripedpajamas** [comment]
Hi @LanceW

This should be fixed in 1.7.2. Can you install and give it a shot?

**Comment by lancew** [comment]
I'm running from the git repo, master latest commit ( 0d0f8a2f75ed22de09590d9022d538908ee8f45a ).

**Comment by stripedpajamas** [comment]
I may have forgotten to push 😩😩😩😩😩😩 my internet is down because of Florence but I'll push up what is currently in 1.7.2 when internet returns

**Comment by lancew** [comment]
Stay safe!

**Comment by stripedpajamas** [comment]
pushed 🎉

**Comment by lancew** [comment]
Latest instance:

```
> @squi|TypeError: Cannot read property 'startsWith' of null
    at actions.authors.get.filter.name (/home/lancew/dev/scat/node_modules/ssb-chat-core/state/actions.js:77:24)
    at /home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:3084:23
    at /home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1377:16
    at ValueNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1744:12)
    at BitmapIndexedNode.iterate.HashArrayMapNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1737:24)
    at HashArrayMapNode.BitmapIndexedNode.iterate.HashArrayMapNode.iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1737:24)
    at Map.__iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:1375:32)
    at KeyedIterable.filterSequence.__iterateUncached (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:3083:16)
    at seqIterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:604:16)
    at KeyedIterable.Seq.__iterate (/home/lancew/dev/scat/node_modules/immutable/dist/immutable.js:274:14)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ssb-chat@1.7.2 start: `node index.js "--debug"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ssb-chat@1.7.2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lancew/.npm/_logs/2018-09-17T17_07_10_300Z-debug.log
[lancew@spectre scat]$ cat /home/lancew/.npm/_logs/2018-09-17T17_07_10_300Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/home/lancew/.nvm/versions/node/v10.0.0/bin/node',
1 verbose cli   '/home/lancew/.nvm/versions/node/v10.0.0/bin/npm',
1 verbose cli   'start',
1 verbose cli   '--',
1 verbose cli   '--debug' ]
2 info using npm@6.4.1
3 info using node@v10.0.0
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle ssb-chat@1.7.2~prestart: ssb-chat@1.7.2
6 info lifecycle ssb-chat@1.7.2~start: ssb-chat@1.7.2
7 verbose lifecycle ssb-chat@1.7.2~start: unsafe-perm in lifecycle true
8 verbose lifecycle ssb-chat@1.7.2~start: PATH: /home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/lancew/dev/scat/node_modules/.bin:/home/lancew/perl5/bin:/home/lancew/.phpenv/bin:/home/lancew/.nvm/versions/node/v10.0.0/bin:/home/lancew/perl5/perlbrew/bin:/home/lancew/perl5/perlbrew/perls/perl-5.28.0/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/android-sdk/platform-tools:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/opt/flutter/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/lancew/.gem/ruby/2.4.0/bin:/home/lancew/.perl6/bin
9 verbose lifecycle ssb-chat@1.7.2~start: CWD: /home/lancew/dev/scat
10 silly lifecycle ssb-chat@1.7.2~start: Args: [ '-c', 'node index.js "--debug"' ]
11 silly lifecycle ssb-chat@1.7.2~start: Returned: code: 1  signal: null
12 info lifecycle ssb-chat@1.7.2~start: Failed to exec start script
13 verbose stack Error: ssb-chat@1.7.2 start: `node index.js "--debug"`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at EventEmitter.emit (events.js:182:13)
13 verbose stack     at ChildProcess.<anonymous> (/home/lancew/.nvm/versions/node/v10.0.0/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:947:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:236:5)
14 verbose pkgid ssb-chat@1.7.2
15 verbose cwd /home/lancew/dev/scat
16 verbose Linux 4.18.7-arch1-1-ARCH
17 verbose argv "/home/lancew/.nvm/versions/node/v10.0.0/bin/node" "/home/lancew/.nvm/versions/node/v10.0.0/bin/npm" "start" "--" "--debug"
18 verbose node v10.0.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 1
22 error ssb-chat@1.7.2 start: `node index.js "--debug"`
22 error Exit status 1
23 error Failed at the ssb-chat@1.7.2 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

This is on master at commit eef4e06ffe96b9815979cc113f5f33f1106b415b

**Comment by stripedpajamas** [comment]
package-lock problem... fixed in 9c8468f4d3ebfce4051f01d1a3cbabaa3e001bbf
